src/serac/physics/tests/serac_thermal_solid_functional.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019-2022, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/thermal_solid_functional.hpp"
+#include "serac/physics/materials/thermal_functional_material.hpp"
+#include "serac/physics/materials/solid_functional_material.hpp"
+
+#include <fstream>
+
+#include <gtest/gtest.h>
+#include "mfem.hpp"
+
+#include "serac/serac_config.hpp"
+#include "serac/mesh/mesh_utils.hpp"
+#include "serac/physics/state/state_manager.hpp"
+
+namespace serac {
+
+template <int p, int dim>
+void functional_test_static(double expected_norm)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  int serial_refinement   = 1;
+  int parallel_refinement = 0;
+
+  // Create DataStore
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, "thermal_functional_static_solve");
+
+  static_assert(dim == 2 || dim == 3, "Dimension must be 2 or 3 for thermal functional test");
+
+  // Construct the appropriate dimension mesh and give it to the data store
+  std::string filename =
+      (dim == 2) ? SERAC_REPO_DIR "/data/meshes/star.mesh" : SERAC_REPO_DIR "/data/meshes/beam-hex.mesh";
+
+  auto mesh = mesh::refineAndDistribute(buildMeshFromFile(filename), serial_refinement, parallel_refinement);
+  serac::StateManager::setMesh(std::move(mesh));
+
+  // Define a boundary attribute set
+  std::set<int> ess_bdr = {1};
+
+  // define the solver configurations
+  auto thermal_options = ThermalConductionFunctional<p, dim>::defaultQuasistaticOptions();
+
+  // no default solver options for solid yet, so make some here
+  const IterativeSolverOptions default_linear_options = {.rel_tol     = 1.0e-6,
+                                                         .abs_tol     = 1.0e-10,
+                                                         .print_level = 0,
+                                                         .max_iter    = 500,
+                                                         .lin_solver  = LinearSolver::GMRES,
+                                                         .prec        = HypreBoomerAMGPrec{}};
+
+  const NonlinearSolverOptions default_nonlinear_options = {
+    .rel_tol = 1.0e-4, .abs_tol = 1.0e-8, .max_iter = 10, .print_level = 1};
+
+  const typename SolidFunctional<p, dim>::SolverOptions solid_options = {default_linear_options,
+    default_nonlinear_options};
+  
+  // Construct a functional-based thermal-solid solver
+  ThermalSolidFunctional<p, dim> thermal_solid_solver(thermal_options,
```

Comment:
I would add a note saying the material call still needs to be implemented before this module can be used.